### PR TITLE
En 10102 unit tests sc execution

### PR DESCRIPTION
**Pull request by raduchis**
more unit tests sc execution added

### CODE REVIEW

File: data/block/blockV2.go [hunk 1]
```diff
@@ -481,6 +481,10 @@ func (hv2 *HeaderV2) SetScheduledRootHash(rootHash []byte) error {
 
 // ValidateHeaderVersion does extra validation for header version
 func (hv2 *HeaderV2) ValidateHeaderVersion() error {
+	if hv2 == nil {
```
Comment (AdoAdoAdo) [inline data/block/blockV2.go:484]: :+1:

File: data/block/blockV2.go [hunk 2]
```diff
@@ -491,6 +495,10 @@ func (hv2 *HeaderV2) ValidateHeaderVersion() error {
 
 // SetAdditionalData sets the additional version related data for the header
 func (hv2 *HeaderV2) SetAdditionalData(headerVersionData headerVersionData.HeaderAdditionalData) error {
+	if hv2 == nil {
```
Comment (AdoAdoAdo) [inline data/block/blockV2.go:498]: :+1:

**Review by AdoAdoAdo — APPROVED** [review]

File: process/block/metablock_test.go [hunk 3]
```diff
@@ -266,6 +294,78 @@ func TestNewMetaProcessor_NilForkDetectorShouldErr(t *testing.T) {
 	assert.Nil(t, be)
 }
 
+func TestNewMetaProcessor_NilEpochStartDataCreatorShouldErr(t *testing.T) {
+	t.Parallel()
+
+	coreComponents, dataComponents, bootstrapComponents, statusComponents := createMockComponentHolders()
+	arguments := createMockMetaArguments(coreComponents, dataComponents, bootstrapComponents, statusComponents)
```
Comment (bogdan-rosianu) [inline process/block/metablock_test.go:301]: could refactor to `createMockMetaArguments(createMockComponentHolders())` in all tests where you don't change a component holder

File: process/block/metablock_test.go [hunk 4]
```diff
@@ -2820,3 +2920,87 @@ func TestMetaProcessor_CreateAndProcessBlockCallsProcessAfterFirstEpoch(t *testi
...
+			return epoch
+		},
+	}
+
+	mp, err := blproc.NewMetaProcessor(arguments)
+	assert.Nil(t, err)
+
+	h, err := mp.CreateNewHeader(round, nonce)
+	assert.Nil(t, err)
+	assert.IsType(t, &block.MetaBlock{}, h)
```
Comment (bogdan-rosianu) [inline process/block/metablock_test.go:2995]: :+1:

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**